pkg/handler/repositories.go

```diff
@@ -299,11 +299,31 @@ func (rh *RepositoryHandler) update(c echo.Context, fillDefaults bool) error {
 	if fillDefaults {
 		repoParams.FillDefaults()
 	}
-	if err := rh.DaoRegistry.RepositoryConfig.Update(orgID, uuid, repoParams); err != nil {
+
+	repoConfig, err := rh.DaoRegistry.RepositoryConfig.Fetch(orgID, uuid)
+	if err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching repository", err.Error())
+	}
+
+	if repoParams.URL != nil && repoConfig.URL != *repoParams.URL {
+		snapInProgress, err := rh.DaoRegistry.TaskInfo.IsSnapshotInProgress(orgID, uuid)
```

Comment:
this takes the repository UUID, not the repositoryConfig UUID, so at the moment I can change the URLs while a snapshot is being created

Reply:
doh, this probably broke on a rebase

Reply:
should be fixed now